lib/ts/recipe/passkey/types.ts

```diff
@@ -0,0 +1,473 @@
+    relyingPartyName?: string | ((input: { request: BaseRequest | undefined; userContext: UserContext }) => string);
+    override?: {
+        functions?: (
+            originalImplementation: RecipeInterface,
+            builder?: OverrideableBuilder<RecipeInterface>
+        ) => RecipeInterface;
+        apis?: (originalImplementation: APIInterface, builder?: OverrideableBuilder<APIInterface>) => APIInterface;
+    };
+};
+
+export type RecipeInterface = {
+    registerPasskeyOptions(input: {
+        email: string;
+        password: string;
+        session: SessionContainerInterface | undefined;
+        tenantId: string;
+        userContext: UserContext;
+    }): Promise<{
+        status: "OK";
+        passkeyGeneratedOptionsId: string;
+        rp: {
+            id: string;
+            name: string;
+        };
+        user: {
+            id: string;
+            name: string;
+            displayName: string;
+        };
+        challenge: string;
+        timeout: number;
+        excludeCredentials: {
+            id: string;
+            type: string;
+            transports: ("ble" | "hybrid" | "internal" | "nfc" | "usb")[];
+        }[];
+        attestation: "none" | "indirect" | "direct" | "enterprise";
+        pubKeyCredParams: {
+            alg: number;
+            type: string;
+        }[];
+        authenticatorSelection: {
+            requireResidentKey: boolean;
+            residentKey: "required" | "preferred" | "discouraged";
+            userVerification: "required" | "preferred" | "discouraged";
+        };
+    }>;
+
+    signInPasskeyOptions(input: {
+        session: SessionContainerInterface | undefined;
+        tenantId: string;
+        userContext: UserContext;
+    }): Promise<{
+        status: "OK";
+        passkeyGeneratedOptionsId: string;
+        challenge: string;
+        timeout: number;
+        userVerification: "required" | "preferred" | "discouraged";
+    }>;
+
+    signUp(input: {
+        email: string | undefined;
+        passkeyGeneratedOptionsId: string;
+        passkey: {
+            id: string;
+            rawId: string;
+            response: {
+                clientDataJSON: string;
+                attestationObject: string;
+                transports?: ("ble" | "cable" | "hybrid" | "internal" | "nfc" | "smart-card" | "usb")[];
+                userHandle: string;
+            };
+            authenticatorAttachment: "platform" | "cross-platform";
+            clientExtensionResults: Record<string, unknown>;
+            type: "public-key";
+        };
+        session: SessionContainerInterface | undefined;
+        shouldTryLinkingWithSessionUser: boolean | undefined;
+        tenantId: string;
+        userContext: UserContext;
+    }): Promise<
+        | {
+              status: "OK";
+              user: User;
+              recipeUserId: RecipeUserId;
+          }
+        | { status: "EMAIL_ALREADY_EXISTS_ERROR" }
```

Comment:
not sure if this is the best way, cause if this error comes, but the user has already added a passkey in their browser, so it would get confusing

Reply:
So instead, we should have this type of error in the part where we are generating a passkeyoption for the purpose of sign up?